src/Trinity.C/src/Storage/MTHash/MTHash.h

```diff
@@ -83,17 +83,35 @@ namespace Storage
             GuardedEntryCount = 1024,
         };
 
+        ///  Memory trunk layout:
+        ///  ,-----------------------------,
+        ///  |                             |
+        ///  |  MemoryTrunk 2GB reserved   |
+        ///  |                             |
+        ///  |---------------+-------------|
+        ///  | r CellEntries | r MTEntries | 
+        ///  |---------------+--+----------|
+        ///  | b BucketPointers | b locks  |
+        ///  `------------------+----------'
+        ///  r = ReserveEntriesPerMTHash
+        ///  b = BucketCount
+        ///  all blocks shown in the chart shall be aligned up to page boundaries.
+
         static uint64_t MTEntryOffset()
         {
-            return TrinityConfig::MemoryReserveUnit() / sizeof(CellEntries[0]);
+            return Memory::RoundUpToPage_64(TrinityConfig::ReserveEntriesPerMTHash() * sizeof(CellEntries[0]));
         }
         static uint64_t BucketMemoryOffset()
         {
-            return MTEntryOffset() + TrinityConfig::MemoryReserveUnit() / sizeof(MTEntries[0]);
+            return Memory::RoundUpToPage_64(MTEntryOffset() + TrinityConfig::ReserveEntriesPerMTHash() * sizeof(MTEntries[0]));
         }
         static uint64_t BucketLockerMemoryOffset()
         {
-            return BucketMemoryOffset() + TrinityConfig::MemoryReserveUnit() / sizeof(Buckets[0]);
+            return Memory::RoundUpToPage_64(BucketMemoryOffset() + BucketCount * sizeof(Buckets[0]));
```

Comment:
Here we changed from `ReserveEntriesPerMTHash` to `BucketCount`. Will it cause any issue?